sg/storage.go

```diff
@@ -0,0 +1,39 @@
+package sg
+
+import (
+	"context"
+
+	"github.com/nspcc-dev/neofs-proto/refs"
+)
+
+type (
+	// Store is a interface for storing users storage group
+	Store interface {
+		Lister
+		Receiver
+		Receptacle
+	}
+
+	// Lister defines list function that returns all storage groups
+	// created for the passed container
+	Lister interface {
+		List(ctx context.Context, cid refs.CID) ([]refs.SGID, error)
+	}
+
+	// Receiver defines get function that returns asked storage group
+	Receiver interface {
+		Get(ctx context.Context, cid refs.CID, sgid refs.SGID) (Provider, error)
+	}
+
+	// Receptacle defines put function that places storage group in the
+	// store.
+	Receptacle interface {
+		Put(ctx context.Context, sg Provider) error
+	}
+
+	// InfoReceiver defines GetSGInfo function that returns storage group
+	// that contains passed object ids.
+	InfoReceiver interface {
```

Comment:
I see issues with this interface. Please check it before merge.

First of all it is not connected with `Store`. It might be confusing to have this interface there, so consider to put it somewhere else. 

More important issue is a return type. Previously it was `SGInfo` struct which contained homomorphic hash and size of the storage group. 
```go
type SGInfo struct {
  ValidationDataSize   uint64 
  ValidationHash       Hash
  . . . 
}
```
Now it is a `Provider` interface, that should provide `IDInfo` for example. The issue is - it is impossible to provide correct `IDInfo` in this function. `IDInfo` contains _one_ storage group id and this function ask to return validation data of _many_ storage groups (`group []refs.ObjectID` in params). For sure we can omit `IDInfo` and return nothing in this case, but it isn't proper `Provider` then. 

I haven't managed these issues when I was looking at storage groups in neofs-proto. Don't forget to look at the `GetSGInfo()` implementation at `lib/implementation/dataaudit.go`, maybe it will help.  Also `GetSGInfo()` is used only once, maybe it is reasonable to keep it in neofs-node, instead of porting it here.

Reply:
@AlexVanin see latest changes